python/pyspark/worker.py

```diff
@@ -80,14 +77,12 @@ def wrap_pandas_udf(f, return_type):
     arrow_return_type = toArrowType(return_type)
 
     def verify_result_length(*a):
-        kwargs = a[-1]
-        result = f(*a[:-1], **kwargs)
-        if len(result) != kwargs["length"]:
+        result = f(*a)
+        if len(result) != len(a[0]):
```

Comment:
I guess we are not guaranteed to have `__len__` in `result`, e.g., `pandas_udf(lambda x: 1, LongType())`. Probably, checking this attribute ahead should be done ahead, while we are here.

Reply:
Should we verify the returned is a Pandas.Series?

Reply:
Good point. We should probably have a test that returns a scalar value too. I'm not sure we should limit the return type so much.  As long as pyarrow can consume it, then it should be ok - it can also take a numpy array which might be useful. Otherwise it should raise a clear exception.  Maybe checking that it has `__len__` is good enough?